thunderfish/psdtype.py

```diff
@@ -0,0 +1,173 @@
+    of the frequency bins (see: get_bin_percentiles()) and the proportions array calculated from these percentiles. With
+    all these arrays this function is plotting what the rest of the modul doing.
+
+    :param freqs:           (1-D array) frequency array of a psd.
+    :param power:           (1-D array) power array of a psd.
+    :param proportions:     (1-D array) proportions of the single psd bins.
+    :param percentiles:     (2-D array) for every bin four values are calulated and stored in separate lists. These four
+                            values are percentiles of the respective bins.
+    :param ax:              (axis for plot) empty axis that is filled with content in the function.
+    :return ax:             (axis for plot) axis that is ready for plotting.
+    """
+    ax.plot(freqs[:int(3000 / (freqs[-1] / len(freqs)))],
+                10.0 * np.log10(power[:int(3000 / (freqs[-1] / len(freqs)))]), '-', alpha=0.5)
+    for bin in np.arange(len(proportions)):
+        ax.fill_between([bin * 125, (bin + 1) * 125], percentiles[0][bin], percentiles[1][bin], color='red',
+                        alpha=0.7)
+        ax.fill_between([bin * 125, (bin + 1) * 125], percentiles[1][bin], percentiles[2][bin], color='green',
+                        alpha=0.7)
+        ax.fill_between([bin * 125, (bin + 1) * 125], percentiles[2][bin], percentiles[3][bin], color='red',
+                        alpha=0.7)
+    ax.set_xlim([0, 3000])
+    ax.set_xlabel('Frequency')
+    ax.set_ylabel('Power [dB]')
+    return ax
+
+def psd_assignment(power, freqs, freq_bins=125, max_freq = 3000, plot_data_func=None, **kwargs):
+    """
+    Function that is called when you got a PSD and want to find out from what fishtype this psd is. It with the help of
+    several other function it analysis the structur of the EOD and can with this approach tell what type of fish the PSD
+    comes from.
+
+    :param power:           (1-D array) power array of a psd.
+    :param freqs:           (1-D array) frequency array of a psd.
+    :param freq_bins:       (float) width of frequency bins in which the psd shall be divided (Hz) [bin_it() function].
+    :param max_freq:        (float) maximum frequency that shall be provided in the separated power array [bin_it() function].
+    :param plot_data_func:  (function) function (psdtypeplot()) that is used to create a axis for later plotting about the process of psd
+                            type detection.
+    :param **kwargs:        additional arguments that are passed to the plot_data_func().
+    :return psd_type:       (string) "wave" or "pulse" depending on the proportion of the psd.
+    :return proportions:    (1-D array) proportions of the single psd bins.
+    :return ax:             (axis for plot) axis that is ready for plotting containing a figure that shows what the modul did.
+
+    """
+    print('\nAssigning PSD-Type ...')
+    res = freqs[-1]/len(freqs) # resolution
```

Comment:
this is wrong! For example
freqs = [0.0, 0.5, 1.0]
then res would be 1.0/3 = 0.33.

What you want is
res = freqs[1] - freqs[0]

or
res = np.mean(np.diff(freqs))